packages/webapp/src/containers/Animals/Inventory/KPI/index.tsx

```diff
@@ -0,0 +1,127 @@
+import { CustomAnimalType, DefaultAnimalType } from '../../../../store/api/types';
+import { ReactComponent as CattleIcon } from '../../../../assets/images/animals/cattle-icon.svg';
+import { ReactComponent as ChickenIcon } from '../../../../assets/images/animals/chicken-icon.svg';
+import { ReactComponent as PigIcon } from '../../../../assets/images/animals/pig-icon.svg';
+import { ReactComponent as CustomIcon } from '../../../../assets/images/nav/animals.svg';
+import { uppercaseTheFirstLetter } from '../../../../util';
+import { getComparator } from '../../../../util/sort';
+import { generateUniqueAnimalTypeId } from '../../../../util/animal';
+import styles from './styles.module.scss';
+
+const icons = {
+  CATTLE: <CattleIcon />,
+  CHICKEN_BROILERS: <ChickenIcon />,
+  CHICKEN_LAYERS: <ChickenIcon />,
+  PIGS: <PigIcon />,
+  '': <CustomIcon />,
+};
+
+const formatAnimalTypes = (
+  types: (DefaultAnimalType | CustomAnimalType)[],
+  onTypeClick: (typeId: string) => void,
+  t: TFunction,
+) => {
+  const formattedTypes: TypeCountTile[] = [];
+  types.forEach((type) => {
+    if (type.count) {
+      const label = 'type' in type ? type.type : t(`animalTypes:${type.key}`);
+      const iconKey = ('key' in type ? type.key : '') as keyof typeof icons;
+      const id = generateUniqueAnimalTypeId(type);
+
+      formattedTypes.push({
+        label: uppercaseTheFirstLetter(label),
+        count: type.count,
+        icon: icons[iconKey],
+        id,
+        onClick: () => onTypeClick(id),
+      });
+    }
+  });
+  return formattedTypes;
+};
+
+interface KPIProps {
+  isCompactSideMenu: boolean;
+  selectedTypeIds: string[];
+  onTypeClick: (typeId: string) => void;
+}
+
+function KPI({ isCompactSideMenu, selectedTypeIds, onTypeClick }: KPIProps) {
+  const [kpiHeight, setKpiHeight] = useState<number | null>(null);
+
+  const { t } = useTranslation(['translation', 'common', 'animalTypes']);
+  const wrapperRef = useRef<HTMLDivElement>(null);
+  const { data, isLoading } = useQueries([
+    { label: 'defaultAnimalTypes', hook: useGetDefaultAnimalTypesQuery, params: '?count=true' },
+    { label: 'customAnimalTypes', hook: useGetCustomAnimalTypesQuery, params: '?count=true' },
+  ]);
+
+  useEffect(() => {
+    const setHeight = () => setKpiHeight(wrapperRef.current?.offsetHeight || null);
+
+    setHeight();
+    window.addEventListener('resize', () => {
+      setHeight();
+    });
+
+    return () => {
+      window.removeEventListener('resize', () => {
+        setHeight();
+      });
+    };
+  }, []);
+
+  const types = useMemo(() => {
+    if (isLoading || !data) {
+      return [];
+    }
+
+    const { defaultAnimalTypes, customAnimalTypes } = data;
+    const types = formatAnimalTypes([...defaultAnimalTypes, ...customAnimalTypes], onTypeClick, t);
+    types.sort(getComparator('asc', 'label'));
+
+    return types;
+  }, [data, isLoading, onTypeClick]);
+
+  return (
+    <div style={{ height: kpiHeight || 0 }}>
```

Comment:
I don't like the inline CSS, but KPI's height changes unpredictably, so I went with the easiest way. Since the child component's position is `fixed`, we need to maintain the height of the KPI component so that the table remains visible and is not hidden beneath the KPI component.

Reply:
Clean solution! Will it make a difference when the table itself has some fixed property (like the header)? I see the table is properly placed with the height correction, but the headers do go under the KPI on scroll -- not sure if this will still be the case with @Duncan-Brain's update?

<img width="635" alt="Screenshot 2024-02-29 at 6 08 31 PM" src="https://github.com/LiteFarmOrg/LiteFarm/assets/82857964/68342526-eb28-45a3-b70c-eaf2be6ee5b5">

Reply:
@kathyavini I am not handling any stickyness in the inventory implementation -- does it say somewhere that table header should be sticky?  I can't find it anywhere.

Reply:
@Duncan-Brain I thought I remember Loïc saying it verbally, something like the "these are fixed as obviously all table headers are" 🤔 Might be misremembering though!